instrumentation/activej-http-6.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/activejhttp/ActivejAsyncServletInstrumentation.java

```diff
@@ -52,39 +54,46 @@ public void transform(TypeTransformer transformer) {
   @SuppressWarnings("unused")
   public static class ServeAdvice {
 
+    public static class AdviceLocals {
+      public HttpRequest httpRequest;
+      public Context context;
+      public Scope scope;
+    }
+
+    @Nullable
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void methodEnter(
-        @Advice.This AsyncServlet asyncServlet,
-        @Advice.Argument(0) HttpRequest request,
-        @Advice.Local("otelContext") Context context,
-        @Advice.Local("otelScope") Scope scope,
-        @Advice.Local("httpRequest") HttpRequest httpRequest) {
+    public static AdviceLocals methodEnter(
+        @Advice.This AsyncServlet asyncServlet, @Advice.Argument(0) HttpRequest request) {
+
       Context parentContext = currentContext();
-      httpRequest = request;
       if (!instrumenter().shouldStart(parentContext, request)) {
-        return;
+        return null;
```

Comment:
@JonasKunz returning `null` here is a manual change done after automatic rewrite to avoid allocation when we can.

Reply:
This is technically not needed for performance reasons I think: If the advice is "hot" and get's inlined, the JVM will very likely perform a [scalar replacement](https://shipilev.net/jvm/anatomy-quarks/18-scalar-replacement/) of the `AdviceLocals` object, as it doesn't escape the instrumented method.

However, for style reasons I think your change is still useful, as the early-out is reached after less noise.